plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryClient.java

```diff
@@ -126,6 +126,18 @@ Table update(TableInfo table)
         return bigQuery.update(table);
     }
 
+    public void createTable(TableInfo tableInfo)
+    {
+        bigQuery.create(tableInfo);
+        tableIds.put(tableInfo.getTableId(), tableInfo.getTableId());
```

Comment:
The `tableIds` map in BigQueryClient doesn't actually store anything useful since both the keys and values are same.

Each `get` on the map is also accompanied by an API call too so we should perform the API call directly.

I've removed this map in https://github.com/trinodb/trino/pull/6748 in favour of a cache which maintains a mapping of lowercase name to remote name.

Reply:
If #6748 is merged first, I would remove this. But, I think we should keep this line for now.

Reply:
Removed relevant code.